examples/src/main/resources/precompile-example/IHederaTokenService.sol

```diff
@@ -0,0 +1,763 @@
+    /// @param sender The sender for the transaction
+    /// @param recipient The receiver of the transaction
+    /// @param amount Non-negative value to send. a negative value will result in a failure.
+    function transferToken(
+        address token,
+        address sender,
+        address recipient,
+        int64 amount
+    ) external returns (int64 responseCode);
+
+    /// Transfers tokens where the calling account/contract is implicitly the first entry in the token transfer list,
+    /// where the amount is the value needed to zero balance the transfers. Regular signing rules apply for sending
+    /// (positive amount) or receiving (negative amount)
+    /// @param token The token to transfer to/from
+    /// @param sender The sender for the transaction
+    /// @param recipient The receiver of the transaction
+    /// @param serialNumber The serial number of the NFT to transfer.
+    function transferNFT(
+        address token,
+        address sender,
+        address recipient,
+        int64 serialNumber
+    ) external returns (int64 responseCode);
+
+    /// Allows spender to withdraw from your account multiple times, up to the value amount. If this function is called
+    /// again it overwrites the current allowance with value.
+    /// Only Applicable to Fungible Tokens
+    /// @param token The hedera token address to approve
+    /// @param spender the account address authorized to spend
+    /// @param amount the amount of tokens authorized to spend.
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function approve(
+        address token,
+        address spender,
+        uint256 amount
+    ) external returns (int64 responseCode);
+
+    /// Returns the amount which spender is still allowed to withdraw from owner.
+    /// Only Applicable to Fungible Tokens
+    /// @param token The Hedera token address to check the allowance of
+    /// @param owner the owner of the tokens to be spent
+    /// @param spender the spender of the tokens
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    /// @return allowance The amount which spender is still allowed to withdraw from owner.
+    function allowance(
+        address token,
+        address owner,
+        address spender
+    ) external returns (int64 responseCode, uint256 allowance);
+
+    /// Allow or reaffirm the approved address to transfer an NFT the approved address does not own.
+    /// Only Applicable to NFT Tokens
+    /// @param token The Hedera NFT token address to approve
+    /// @param approved The new approved NFT controller.  To revoke approvals pass in the zero address.
+    /// @param serialNumber The NFT serial number  to approve
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function approveNFT(
+        address token,
+        address approved,
+        int64 serialNumber
```

Comment:
```suggestion
        uint256 serialNumber
```
like it is in [IHederaTokenService.sol](https://github.com/hashgraph/hedera-smart-contracts/blob/8546133775642df03ab63d7d7141ef560c41da29/contracts/hts-precompile/IHederaTokenService.sol#L526) in the smart-contracts repo (the last arg has a different type and this changes the function signature).